src/ucp/wireup/wireup.c

```diff
@@ -124,6 +125,11 @@ ucs_status_t ucp_wireup_msg_progress(uct_pending_req_t *self)
 
     /* send the active message */
     req->send.lane = ucp_wireup_get_msg_lane(ep, req->send.wireup.type);
+    uct_ep         = ep->uct_eps[req->send.lane];
+
+    if (!ucp_wireup_ep_test(uct_ep)) {
+        ucs_assert(req->send.wireup.type != UCP_WIREUP_MSG_PRE_REQUEST);
+    }
```

Comment:
maybe move to separate function and assert all allowed types there?
This condition hardly to read and might  be broken by new WIREUP_MSG type. Also probably it could be reused above.

Reply:
> maybe move to separate function and assert all allowed types there?
> This condition hardly to read and might be broken by new WIREUP_MSG type.

we disallow sending `PRE_REQUEST` only.
adding new packet type will broke it too, if it will be allowed to be sent.

Reply:
removed this assertion, because it is not needed